Sources/swift-build-prebuilts/main.swift

```diff
@@ -0,0 +1,437 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2014-2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Checkout supported versions of the prebuilts, build them, package up the zip and update the manifest file.
+
+// Ideally this would be a script, however until we have package dependencies
+// for scripts, this will need to be a part of the package. But this is why it's
+// reading like a script
+
+import Basics
+import Foundation
+import TSCBasic
+import Workspace
+
+struct PrebuiltRepos: Identifiable {
+    let url: URL
+    let versions: [Version]
+
+    var id: URL { url }
+
+    struct Version: Identifiable {
+        let tag: String
+        let manifest: Workspace.PrebuiltsManifest
+        let cModulePaths: [String: [String]]
+
+        var id: String { tag }
+    }
+}
+
+var prebuiltRepos: IdentifiableSet<PrebuiltRepos> = [
+    .init(
+        url: .init(string: "https://github.com/swiftlang/swift-syntax")!,
+        versions: [
+            .init(
+                tag:"600.0.1",
+                manifest: .init(libraries: [
+                    .init(
+                        name: "MacroSupport",
+                        products: [
+                            "SwiftSyntaxMacrosTestSupport",
+                            "SwiftCompilerPlugin",
+                            "SwiftSyntaxMacros"
+                        ],
+                        cModules: [
+                            "_SwiftSyntaxCShims",
+                        ]
+                    ),
+
+                ]),
+                cModulePaths: [
+                    "_SwiftSyntaxCShims": ["Sources", "_SwiftSyntaxCShims"]
+                ]
+            ),
+        ]
+    ),
+]
+
+let manifestHost = URL(string: "https://github.com/dschaefer2/swift-syntax/releases/download")!
+
+let fm = FileManager.default
+let args = ProcessInfo.processInfo.arguments
+
+let swiftVersion = "\(SwiftVersion.current.major).\(SwiftVersion.current.minor)"
+
+var stageDir = Path(fm.currentDirectoryPath + "/stage")
+
+var docker = false
+var dockeronly = false
+var dockercmd = "docker"
+let dockerImageRoot = "swiftlang/swift:nightly-"
+
+var i = 1
+while i < args.count {
```

Comment:
Is there a reason this isn't structured as a 'build tool' like swift-bootstrap? Specifically, using ArgumentParser to document the flags and to structure the execution.

Reply:
It's a script. It's not part of the toolchain and is not user facing. It'll only be used to produce the prebuilt binaries in CI. That said, half way through I started to yearn from the argument parser. We can save that for a future iteration.

Reply:
You can still make it a target in the package without it becoming part of the toolchain. The toolchain SwiftPM gets built with CMake anyways, so adding a target to the Package.swift won't have any impact.

Reply:
It is a target in the package. We don't have real support for swift scripts with package dependencies yet so I have to do it this way. It is on our roadmap though.

Reply:
OK, so it is a target in the package. And actually I see it is importing other modules from the package already. So I'm not quite following why you can't use ArgumentParser and TSCBasic.Path as there should be no blocker to doing so.

Reply:
True.